### PR TITLE
Data collection 2

### PR DESCRIPTION
**Pull request by mroz22**
- add analyticsReducer
- save instanceId into storage, maybe @slowbackspace check if added correctly
- useAnalytics hook which decorates any method with dispatch(analyticsActions.report()) (discussed with @szymonlesisz)
- added bunch of analytics events

### CODE REVIEW

File: packages/suite/src/actions/suite/analyticsActions.ts [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
-    { type: 'transport-type'; payload: { type: string; version: string } }
+export type AnalyticsActions =
+    | { type: typeof ANALYTICS.DISPOSE }
+    | { type: typeof ANALYTICS.INIT; payload: { sessionId: string; instanceId: string } };
+
+export type Payload =
+    | {
+          type: 'suite-ready';
+          payload: {
+              language: ReturnType<GetState>['suite']['settings']['language'];
```
Comment (szymonlesisz) [inline packages/suite/src/actions/suite/analyticsActions.ts]: why not `AppState['suite']...` ?

File: packages/suite/src/middlewares/suite/analyticsMiddleware.ts [hunk 2]
```diff
@@ -18,11 +19,26 @@ import * as analyticsActions from '@suite-actions/analyticsActions';
 const analytics = (api: MiddlewareAPI<Dispatch, AppState>) => (next: Dispatch) => (
     action: Action,
 ): Action => {
+    let state = api.getState();
```
Comment (szymonlesisz) [inline packages/suite/src/middlewares/suite/analyticsMiddleware.ts]: why?

File: packages/suite/src/components/suite/Settings/components/Analytics.tsx [hunk 3]
```diff
@@ -11,16 +11,17 @@ import * as analyticsActions from '@suite-actions/analyticsActions';
...
 const mapDispatchToProps = (dispatch: Dispatch) => ({
-    toggleAnalytics: bindActionCreators(analyticsActions.toggleAnalytics, dispatch),
+    init: bindActionCreators(analyticsActions.init, dispatch),
+    dispose: bindActionCreators(analyticsActions.dispose, dispatch),
 });
 
 export type Props = ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>;
 
-const Analytics = ({ locks, analytics, toggleAnalytics }: Props) => {
+const Analytics = ({ locks, enabled, init, dispose }: Props) => {
```
Comment (szymonlesisz) [inline packages/suite/src/components/suite/Settings/components/Analytics.tsx]: inspiration for version 3 :)

you can have init, dispose and status served from `useAnalitics` hook, locks also are in hooks now, this way you will not need to use Container

File: packages/suite/src/middlewares/suite/analyticsMiddleware.ts (hunk 2 above)
Comment (mroz22) [reply packages/suite/src/middlewares/suite/analyticsMiddleware.ts]: remnant from some experiments, tnx

File: packages/suite/src/actions/suite/analyticsActions.ts (hunk 1 above)
Comment (mroz22) [reply packages/suite/src/actions/suite/analyticsActions.ts]: sure

File: packages/suite/src/components/suite/Settings/components/Analytics.tsx (hunk 3 above)
Comment (mroz22) [reply packages/suite/src/components/suite/Settings/components/Analytics.tsx]: check it now.

**Pull request merged**